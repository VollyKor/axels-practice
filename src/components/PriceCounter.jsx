import ListGroupItem from 'react-bootstrap/ListGroupItem';

import {
    Item,
    Title,
    Price,
    Total,
    TotalPrice,
    Conditions,
} from '../styled/PriceCounter';

export default function PriceCounter() {
    return (
        <>
            <ListGroupItem className="mb-3 px-0 bg-transparent">
                <Item>
                    <Title>Subtotal</Title>
                    <Price>${398}</Price>
                </Item>

                <Item>
                    <Title>Shipping</Title>
                    <Price>{'Free'}</Price>
                </Item>

                <Item>
                    <Title>Taxes</Title>
                    <Price> ${12.12}</Price>
                </Item>
            </ListGroupItem>

            <Item className="d-flex justify-content-between p-0">
                <Total>Total</Total>
                <TotalPrice>${410.12}</TotalPrice>
            </Item>

            <Conditions />
        </>
    );
}
