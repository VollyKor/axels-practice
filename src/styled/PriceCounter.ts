import styled from 'styled-components';

export const Item = styled.div`
    display: flex;
    justify-content: space-between;

    font-size: 0.8rem;
    color: var(--accent-text-color);
`;

export const Title = styled.p`
    margin-bottom: 3px;
`;

export const Price = styled(Title)``;

export const Total = styled(Item)`
    color: var(--main-text-color);
    font-weight: 700;
`;

export const TotalPrice = styled(Total)``;
