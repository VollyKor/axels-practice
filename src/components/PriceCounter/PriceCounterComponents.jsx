import styled from "styled-components";

export const Item = styled.div`
  display: flex;
  justify-content: space-between;

  font-size: 0.8rem;

  color: #757575;
`;

export const Title = styled.p`
  margin-bottom: 3px;
`;

export const Price = styled(Title)``;

export const Total = styled(Item)`
  color: #5a1094;
  font-weight: 700;
`;

export const TotalPrice = styled(Total)``;

const CinditionText = styled.p`
  position: absolute;
  bottom: 0;

  text-align: center;
  font-size: 0.6rem;

  color: #757575;
`;
const ConditionLink = styled.a`
  font-size: 0.65rem;
  color: #757575;
`;

export function Conditions() {
  return (
    <CinditionText className="ml-2">
      All puraches are subject to our{" "}
      <ConditionLink href="/" target="_blanc">
        Terms and Conditions
      </ConditionLink>
    </CinditionText>
  );
}
