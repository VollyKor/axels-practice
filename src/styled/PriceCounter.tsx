import styled from 'styled-components';

export const Item = styled.div`
    display: flex;
    justify-content: space-between;

    font-size: 0.8rem;
    color: var(--accent-text-color);
`;

export const Title = styled.p`
    margin-bottom: 3px;
`;

export const Price = styled(Title)``;

export const Total = styled(Item)`
    color: var(--main-text-color);
    font-weight: 700;
`;

export const TotalPrice = styled(Total)``;

const CinditionText = styled.p`
    bottom: 0;
    position: initial;
    text-align: center;
    font-size: 0.6rem;

    @media screen and (min-width: 992px) {
        position: absolute;
        margin-top: 0.8rem;
    } ;
`;
const ConditionLink = styled.a`
    font-size: 0.65rem;
    color: var(--text-accent-color);
`;

export function Conditions() {
    return (
        <CinditionText className="ml-2 text-muted">
            All puraches are subject to our{' '}
            <ConditionLink href="/" target="_blanc" className="text-muted">
                Terms and Conditions
            </ConditionLink>
        </CinditionText>
    );
}
