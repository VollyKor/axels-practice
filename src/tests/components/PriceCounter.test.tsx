import { shallow } from 'enzyme';

import PriceCounter from 'components/PriceCounter';

it('renders snapshot', () => {
    const wrapper = shallow(<PriceCounter price={20} />);

    expect(wrapper).toMatchSnapshot();
});

it('render props correctly', () => {
    const price = 20;

    const wrapper = shallow(<PriceCounter price={price} />);

    const text = wrapper.text();
    expect(text).toContain(`Taxes${price}`);
});
