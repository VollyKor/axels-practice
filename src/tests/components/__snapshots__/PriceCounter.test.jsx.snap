// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <ListGroupItem
    active={false}
    className="mb-3 px-0 bg-transparent"
    disabled={false}
  >
    <styled.div>
      <styled.p>
        Subtotal
      </styled.p>
      <Styled(styled.p)>
        $
        398
      </Styled(styled.p)>
    </styled.div>
    <styled.div>
      <styled.p>
        Shipping
      </styled.p>
      <Styled(styled.p)>
        Free
      </Styled(styled.p)>
    </styled.div>
    <styled.div>
      <styled.p>
        Taxes
      </styled.p>
      <Styled(styled.p)>
        20
      </Styled(styled.p)>
    </styled.div>
  </ListGroupItem>
  <styled.div
    className="d-flex justify-content-between p-0"
  >
    <Styled(styled.div)>
      Total
    </Styled(styled.div)>
    <Styled(Styled(styled.div))
      className="mb-4"
    >
      $
      410.12
    </Styled(Styled(styled.div))>
  </styled.div>
  <Conditions />
</Fragment>
`;

exports[`renders snapshot 1`] = `
<Fragment>
  <ListGroupItem
    active={false}
    className="mb-3 px-0 bg-transparent"
    disabled={false}
  >
    <styled.div>
      <styled.p>
        Subtotal
      </styled.p>
      <Styled(styled.p)>
        $
        398
      </Styled(styled.p)>
    </styled.div>
    <styled.div>
      <styled.p>
        Shipping
      </styled.p>
      <Styled(styled.p)>
        Free
      </Styled(styled.p)>
    </styled.div>
    <styled.div>
      <styled.p>
        Taxes
      </styled.p>
      <Styled(styled.p)>
        20
      </Styled(styled.p)>
    </styled.div>
  </ListGroupItem>
  <styled.div
    className="d-flex justify-content-between p-0"
  >
    <Styled(styled.div)>
      Total
    </Styled(styled.div)>
    <Styled(Styled(styled.div))
      className="mb-4"
    >
      $
      410.12
    </Styled(Styled(styled.div))>
  </styled.div>
  <Conditions />
</Fragment>
`;
